### PR TITLE
prevent recursion with log events

### PR DESCRIPTION
**Pull request by icewind1991**
Prevent the event handling of a log event from creating more log events and causing infinite recursion.

### CODE REVIEW

File: lib/private/Log.php [hunk 1]
```diff
@@ -210,15 +210,21 @@ public function debug(string $message, array $context = []) {
...
 		$entry = $this->interpolateMessage($context, $message);
 
-		if ($this->eventDispatcher) {
-			$this->eventDispatcher->dispatchTyped(new BeforeMessageLoggedEvent($app, $level, $entry));
+		if ($this->eventDispatcher && !$inLogEvent) {
+			$inLogEvent = true;
+			try {
+				$this->eventDispatcher->dispatchTyped(new BeforeMessageLoggedEvent($app, $level, $entry));
+			} catch (Throwable $e) {
+				// ignore exceptions from inside the logger, as trying to log them would lead to recursion
```
Comment (come-nc) [inline lib/private/Log.php:226]: But that means a syntax error in BeforeMessageLoggedEvent class would go unnoticed, no?

Comment (icewind1991) [reply lib/private/Log.php:226]: I guess, but that seems better than breaking any logging from such an error.

(also static analysis would detect that specific error)

Comment (icewind1991) [reply lib/private/Log.php:226]: True, but an exception in the logging would still cause issues. We have similar logic in other logging places iirc

**Review by come-nc — APPROVED** [review]

**Comment by skjnldsv** [comment]
Conflicts

**Pull request closed without merging**